src/aap_eda/services/ruleset/activate_rulesets.py

```diff
@@ -135,28 +128,32 @@ def activate(
                     activation_instance=instance,
                     decision_environment=activation.decision_environment,
                 )
-                instance.status = ActivationStatus.COMPLETED
             else:
                 raise ActivationException(f"Unsupported {deployment_type}")
 
-            if str(instance.status) == ActivationStatus.COMPLETED.value:
+            if instance.status == ActivationStatus.COMPLETED:
```

Comment:
I feel previous statement `if str(instance.status) == ActivationStatus.COMPLETED.value` is safer.

Reply:
No need to convert to str, enum comparison is ok (https://docs.python.org/3/howto/enum.html#comparisons)

Reply:
We could extend [our enum classes](https://github.com/ansible/eda-server/blob/main/src/aap_eda/core/enums.py#L74) to inherit from str class, that would allow us to make safe comparisions with strings if we suspect that the value might be just an string. 
```
class ActivationStatus(str, DjangoEnum):
...
```

In that case `instance.status == ActivationStatus.COMPLETED:` will return true if instance.status is a string or a enum value. 

This would be equivalent to [StrEnum class](https://docs.python.org/3/library/enum.html#enum.StrEnum) included in python3.11

Reply:
We have many defined DjangoEnum classes, should we change them all? or only change `ActivationStatus`?

Reply:
Well, it was just an idea, not a requirement for this PR. IMO it would be weird to do it only on one class because the rest are in the same situation. Maybe is better to do it in another PR .

Reply:
@Alex-Izquierdo @bzwei So far keep it as it is, will do it in another PR if necessary